patches/chrome-installer-mini_installer-BUILD.gn.patch

```diff
@@ -1,22 +1,43 @@
 diff --git a/chrome/installer/mini_installer/BUILD.gn b/chrome/installer/mini_installer/BUILD.gn
-index 4e6af9a522de53d458664afe714961f3f246d9b7..ec52e00e7d08ded7368e9f5a92a88eca4d786922 100644
+index 4e6af9a522de53d458664afe714961f3f246d9b7..90d1f6cc0478723b88f34e609348511b1537dd63 100644
 --- a/chrome/installer/mini_installer/BUILD.gn
 +++ b/chrome/installer/mini_installer/BUILD.gn
-@@ -13,6 +13,7 @@ import("//third_party/icu/config.gni")
- import("//tools/v8_context_snapshot/v8_context_snapshot.gni")
- import("//ui/base/ui_features.gni")
- import("//v8/gni/v8.gni")
-+import("//brave/build/config.gni")
+@@ -113,7 +113,7 @@ group("setup_runtime_deps") {
  
- declare_args() {
-   # The Chrome archive is maximally compressed in official builds to reduce the
-@@ -171,6 +172,9 @@ action("mini_installer_archive") {
-     "//mojo/core:shared_library",
-     "//third_party/icu:icudata",
+ packed_files_rc_file = "$target_gen_dir/mini_installer/packed_files.rc"
+ 
+-action("mini_installer_archive") {
++import("//brave/chromium_src/chrome/installer/mini_installer/build.gni") brave_mini_installer_archive_action("mini_installer_archive") {
+   script = "//chrome/tools/build/win/create_installer_archive.py"
+ 
+   release_file = "chrome.release"
+@@ -223,10 +223,10 @@ action("mini_installer_archive") {
+   ]
+ }
+ 
+-executable("mini_installer") {
++template("mini_installer") { executable(target_name) {
+   sources = [
+     "mini_installer_exe_main.cc",
+-    packed_files_rc_file,
++    invoker.packed_files_rc_file,
    ]
-+  if (skip_signing) { args += [ "--skip_signing" ] }
-+  if (build_delta_installer) { args += [ "--setup_exe_format=DIFF", "--diff_algorithm=COURGETTE", "--last_chrome_installer",rebase_path("$root_out_dir/$last_chrome_installer", root_build_dir) ] }
-+  deps += [ "//brave:mini_installer_dependencies" ]
  
-   if (enable_hidpi) {
-     args += [ "--enable_hidpi=1" ]
+   # This target is special so we manually override most linker flags and
+@@ -261,6 +261,7 @@ executable("mini_installer") {
+     "//base:clang_profiling_buildflags",
+     "//build/win:default_exe_manifest",
+   ]
++  if (defined(invoker.extra_deps)) { deps += invoker.extra_deps }
+ 
+   # In general, mini_installer tries to avoid depending on the C++ standard
+   # library for size reasons. This is achieved by:
+@@ -280,7 +281,7 @@ executable("mini_installer") {
+     no_default_deps = true
+     ldflags += [ "/ENTRY:MainEntryPoint" ]
+   }
+-}
```

Comment:
```
sources -= [packed_files_rc_file] sources += [invoker.packed_files_rc_file] if (defined(invoker.extra_deps)) { deps += invoker.extra_deps }
```
?

Reply:
Resolved in 8bbe18ccc57b313c1673e210580776d50c40c87f.

Reply:
Correction: Simplified in cd0a432676b700cb39df68469ed68d36a841efa4.